packages/vitest/src/node/error.ts

```diff
@@ -32,7 +32,8 @@ export async function printError(error: unknown, ctx: Vitest) {
   printErrorMessage(e, ctx.console)
   await printStack(ctx, stacks, nearest, async(s, pos) => {
     if (s === nearest && nearest) {
-      const sourceCode = await fs.readFile(nearest.file, 'utf-8')
+      const file = nearest.sourcePos && nearest.sourcePos.source && nearest.sourcePos.source !== '.' ? join(nearest.file, '../', nearest.sourcePos.source) : nearest.file
```

Comment:
We can use `?.` here, and I think it will be good to extract a `fileFromSourcePosition(position)` utility so it can be reused in `printStack`
```suggestion
      const file = nearest?.sourcePos?.source !== '.' ? join(nearest.file, '../', nearest.sourcePos.source) : nearest.file
```

Reply:
Extracted as: `fileFromParsedStack`, any idea why some tests appear to be failing?

Reply:
Thanks! The errors look unrelated, it seems an issue with a test expectation.

Reply:
Solved